modules/azure/loadbalancer.go

```diff
@@ -0,0 +1,217 @@
+// This function would fail the test if there is an error.
+func GetLoadBalancerFrontendIPConfigNames(t testing.TestingT, loadBalancerName string, resourceGroupName string, subscriptionID string) []string {
+	configName, err := GetLoadBalancerFrontendIPConfigNamesE(loadBalancerName, resourceGroupName, subscriptionID)
+	require.NoError(t, err)
+	return configName
+}
+
+// GetLoadBalancerFrontendIPConfigNamesE ConfigNamesE gets a list of the Frontend IP Configuration Names for the Load Balancer.
+func GetLoadBalancerFrontendIPConfigNamesE(loadBalancerName string, resourceGroupName string, subscriptionID string) ([]string, error) {
+	lb, err := GetLoadBalancerE(loadBalancerName, resourceGroupName, subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get the Frontend IP Configurations
+	lbProps := lb.LoadBalancerPropertiesFormat
+	feConfigs := *lbProps.FrontendIPConfigurations
+	if len(feConfigs) == 0 {
+		// No Frontend IP Configuration present
+		return nil, nil
+	}
+
+	// Get the names of the Frontend IP Configurations present
+	configNames := make([]string, len(feConfigs))
+	for i, config := range feConfigs {
+		configNames[i] = *config.Name
+	}
+
+	return configNames, nil
+}
+
+// GetIPOfLoadBalancerFrontendIPConfig gets the IP and IPType for the specified Load Balancer Frontend IP Configuration.
+// This function would fail the test if there is an error.
+func GetIPOfLoadBalancerFrontendIPConfig(t testing.TestingT, feConfigName string, loadBalancerName string, resourceGroupName string, subscriptionID string) (ipAddress string, publicOrPrivate IPType) {
+	ipAddress, ipType, err := GetIPOfLoadBalancerFrontendIPConfigE(feConfigName, loadBalancerName, resourceGroupName, subscriptionID)
+	require.NoError(t, err)
+	return ipAddress, ipType
+}
+
+// GetIPOfLoadBalancerFrontendIPConfigE gets the IP and IPType for the specified Load Balancer Frontend IP Configuration.
+func GetIPOfLoadBalancerFrontendIPConfigE(feConfigName string, loadBalancerName string, resourceGroupName string, subscriptionID string) (ipAddress string, publicOrPrivate IPType, err1 error) {
+	// Get the specified Load Balancer Frontend Config
+	feConfig, err := GetLoadBalancerFrontendIPConfigE(feConfigName, loadBalancerName, resourceGroupName, subscriptionID)
+	if err != nil {
+		return "", -1, err
+	}
+
+	// Get the Properties of the Frontend Configuration
+	feProps := *feConfig.FrontendIPConfigurationPropertiesFormat
+
+	// Check for the Public Type Frontend Config
+	if feProps.PublicIPAddress != nil {
+		// Get PublicIPAddress resource name from the Load Balancer Frontend Configuration
+		pipName := GetNameFromResourceID(*feProps.PublicIPAddress.ID)
+
+		// Get the Public IP of the PublicIPAddress
+		ipValue, err := GetIPOfPublicIPAddressByNameE(pipName, resourceGroupName, subscriptionID)
+		if err != nil {
+			return "", -1, err
+		}
+
+		return ipValue, IPType(PublicIP), nil
```

Comment:
No need to cast to `IPType` since `PublicIP` is a const of that type already.

Reply:
Updated